Bundle maps into build artifacts

StenAL opened the pull request:
### Description
With this change, the server no longer needs to be deployed alongside a
tracks directory and the symlink to the tracks directory can also be
removed, making the server work without any changes on Windows.

Tracks are now bundled into the jar file generated by `mvn package`,
however the location where tracks are searched for can still be
overridden with a CLI flag.

This required changing the way FileSystemStatsManager and
FileSystemTrackManager load tracks by making it possible to configure
which kind of file system they should use as jar resources can only be
read as files when using a bespoke FileSystem.

Also a bunch of duplicate unused assets have been removed. See individual commits for more details.

### Testing
- ran Maven's `exec:java` task from IntelliJ, which successfully loaded maps from the resources path
- ran jar using CLI (`java -jar server/target/server-2.1.2.0-BETA.jar`), which successfully loaded bundled maps
- ran jar using CLI with `--tracks-dir` argument (`java -jar server/target/server-2.1.2.0-BETA.jar  --tracks-dir server/src/main/resources/tracks`), which successfully loaded maps from my filesystem

PhilippvK commented:
@pehala FIY

pehala commented on server/src/main/java/org/moparforia/server/Server.java:
```diff
@@ -31,14 +39,16 @@
     private ChannelGroup allChannels = new DefaultChannelGroup();
-    private ConcurrentLinkedQueue<Event> events = new ConcurrentLinkedQueue<Event>();
-    private HashMap<PacketType, ArrayList<PacketHandler>> packetHandlers = new HashMap<PacketType, ArrayList<PacketHandler>>();
+    private ConcurrentLinkedQueue<Event> events = new ConcurrentLinkedQueue<>();
```
why change this?

StenAL replied:
Since I was touching this code anyway, I figured I might as well clean it up. The extra generic parameters on the right hand side of the assignment add no extra information but is just more verbose and IntelliJ was complaining about it:
![image](https://github.com/PhilippvK/playforia-minigolf/assets/21343173/4880d446-3860-4dcc-b6b9-4ec6a050806b)

pehala commented on server/src/main/java/org/moparforia/server/Server.java:
```diff
@@ -217,4 +223,36 @@ public void run() {
+     * 3. Else, if running in an IDE, use resources folder
+     * 4. Else, use default filesystem and look in default directory
+     */
+    private TracksLocation getTracksLocation() throws URISyntaxException, IOException {
```
Is it still possible to add custom tracks to the server and if so how?

StenAL replied:
Yes, by using the CLI flag `--tracks-dir` when starting the server. I'll add some clarification around this to the README.

StenAL commented:
Changes since last review:

- clarified how to use custom tracks in README
- fixed CI
  - removed the `assembly-tracks` build step in the `shared` project which created a zip file of all tracks. This is not needed anymore because tracks are now bundled into the jar file.
  - removed the `Upload Tracks` step from CI

PhilippvK commented:
Look fine for me. If I receive no complaints until Monday this gets merged!
